exist-core/src/test/resources-filtered/conf.xml

```diff
@@ -929,7 +929,7 @@
             containing valid nodes.
         
     -->
-    <xupdate allowed-fragmentation="50000" enable-consistency-checks="no"/>
```

Comment:
You don't need to modify the config for every test, and neither should you! You can simply override this in your JUnit test that checks for this by passing configuration properties to the constructor of your EmbeddedExistServer or EmbeddedExistXmldbServer.

Reply:
I do feel it would be good to run our tests in a way that will ensure they work even when they trigger a defragmentation run.

Reply:
Changing the behaviour of every test for this single purpose is not appropriate.

Reply:
It serves a single purpose: highlight all areas where the defragmentation of split documents will cause problems.
And as you can see from the list of failing tests XQueryTrigger might still be affected and this needs further investigation.

Reply:
> It serves a single purpose

Quite the opposite. It mixes concerns... and it will trigger unnecessary defragmentation in all other tests that cause a page slit. This means side-effects and slow-down in the test suite. It isn't needed. Please just add a test for your issue.

Reply:
The thing is, creating a single test-case by creating an instance of EmbeddedExistXmldbServer with the changed setting - as you proposed - was my first attempt. The Properties one uses to set fragmentation limit must be of type string and cannot be an integer value. They are also not casted to an integer where needed. 

Conclusion: It does not work that way.

Reply:
> The Properties one uses to set fragmentation limit must be of type string and cannot be an integer value

@line-o That's not correct, you can set any type of value.

> Conclusion: It does not work that way.

I am not sure what you tried as I can't see your code. I created a quick skeleton for you that works just fine:

```java
package org.exist.storage.btree;

import org.exist.storage.BrokerPool;
import org.exist.test.ExistEmbeddedServer;
import org.exist.util.Configuration;
import org.junit.Before;
import org.junit.ClassRule;
import org.junit.Test;

import java.util.Properties;

import static org.exist.storage.DBBroker.PROPERTY_XUPDATE_FRAGMENTATION_FACTOR;

public class PagedDefragmentationTest {

  private static final Properties configProperties = new Properties();
  static {
    configProperties.put(PROPERTY_XUPDATE_FRAGMENTATION_FACTOR, Integer.valueOf(-1));
  }

  @ClassRule
  public static final ExistEmbeddedServer existEmbeddedServer = new ExistEmbeddedServer(configProperties, true, true);

  // NOTE(AR) just temporary to show that config of Integer properties works within a test context
  @Before
  public void printStatus() {
    final BrokerPool brokerPool = existEmbeddedServer.getBrokerPool();
    System.out.println("Operational=" + brokerPool.isOperational());
    System.out.println("Configured=" + brokerPool.isInstanceConfigured());
    final Configuration configuration = brokerPool.getConfiguration();
    final Object configuredValue = configuration.getProperty(PROPERTY_XUPDATE_FRAGMENTATION_FACTOR);
    System.out.println("Configuration: " + PROPERTY_XUPDATE_FRAGMENTATION_FACTOR + "=" + configuredValue + " (type: " + configuredValue.getClass().getSimpleName() + ")");
  }

  @Test
  public void yourTest() {
    // TODO
  }

}

```

Reply:
Thanks, I was missing the put method on the properties. The test is now in place.
Regardless, I still am convinced that XQueryTriggerTests do indicate there is a problem when defragmentation is triggered.